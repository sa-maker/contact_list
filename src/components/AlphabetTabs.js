import React from 'react';
import TabButton from '../components/TabButton'
import './AlphabetTabs.css';

class AlphabetTabs extends React.Component{

  render() {
    const {contacts, selected, setSelected} = this.props
    return (
      <div className="tabContainer">
        { Object.entries(contacts).map((t,k) => 
          <TabButton 
            value={t[0]} 
            length={t[1].length} 
            extraButton={typeof contacts['_'] !== 'undefined'} 
            selected={selected===t[0]}
            setSelected={setSelected.bind(this) }
            key={k}
          />
        )}
      </div>
    )
  }
}

export default AlphabetTabs;
