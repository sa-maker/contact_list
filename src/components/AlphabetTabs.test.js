import React from 'react'
import renderer from 'react-test-renderer'
import AlphabetTabs from './AlphabetTabs'

test('Open Card displays props', () => {
    const contact_list = {
        a: [1,2,3],
        b: [],
        c: [2451, 447],
        d: ['a','s','d']
        }

    const selected = 'a'
    const setSelected = (val)=>{ return val }

    const component = renderer.create(
      <AlphabetTabs contacts={contact_list} selected={selected} setSelected={setSelected} />  
    )

    let testTabs = component.toJSON()
    expect(testTabs).toMatchSnapshot()
  })