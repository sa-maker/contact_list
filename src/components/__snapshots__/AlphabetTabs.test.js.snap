// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Open Card displays props 1`] = `
<div
  className="tabContainer"
>
  <button
    className="tabButton selected "
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="buttonValue"
    >
      a
    </span>
    <span
      className="buttonLength"
    >
      3
    </span>
  </button>
  <button
    className="tabButton "
    disabled={true}
    onClick={[Function]}
  >
    <span
      className="buttonValue"
    >
      b
    </span>
    <span
      className="buttonLength"
    >
      0
    </span>
  </button>
  <button
    className="tabButton "
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="buttonValue"
    >
      c
    </span>
    <span
      className="buttonLength"
    >
      2
    </span>
  </button>
  <button
    className="tabButton "
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="buttonValue"
    >
      d
    </span>
    <span
      className="buttonLength"
    >
      3
    </span>
  </button>
</div>
`;
